Engine/Module/ModuleResourceManager.cpp

```diff
@@ -39,136 +79,244 @@ update_status ModuleResourceManager::PreUpdate()
 	return update_status::UPDATE_CONTINUE;
 }
 
- bool ModuleResourceManager::CleanUp()
+bool ModuleResourceManager::CleanUp()
 {
-	 thread_comunication.stop_thread = true;
-	 importing_thread.join();
+	thread_comunication.stop_thread = true;
+	importing_thread.join();
 	return true;
 }
 
  void ModuleResourceManager::StartThread()
  {
-	 bool success = App->filesystem->CreateMountedDir("Library");
-	 bool force_import = !App->filesystem->Exists("Library");
-	 if (force_import && !success)
-	 {
-		 return;
-	 }
-
 	 thread_comunication.finished_loading = false;
-	 thread_comunication.total_items = App->filesystem->assets_file->total_sub_files_number;
```

Comment:
I was using this to reimport everything is the library was removed. How are you handling that now?

Reply:
`Library` folder is created in `ModuleFileSystem` if it not exists. After that, the resource manager checks metafiles. If in one metafile the imported file exists but the exported one doesn't, it's imported again